starlette/endpoints.py

```diff
@@ -17,6 +17,10 @@ def __init__(self, scope: Scope, receive: Receive, send: Send) -> None:
         self.scope = scope
         self.receive = receive
         self.send = send
+        self._allowed_methods = filter(
+            lambda method: getattr(self, method.lower(), None) is not None,
+            ["GET", "HEAD", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"],
+        )
```

Comment:
I *guess* I probably find a list comprehension style easier to understand...

```python
self._allowed_methods = [
    method for method in HTTP_METHODS
    if getattr(self, method.lower(), None) is not None
]
```

Butttttt it's just a matter of preference isn't it. Possibly useful feedback(?), but not a blocker to me.

Reply:
yep makes sense, thanks for the suggestion!